StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1024CodeFixProvider.cs

```diff
@@ -0,0 +1,131 @@
+            ImmutableArray.Create(SA1024ColonsMustBeSpacedCorrectly.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (!diagnostic.Id.Equals(SA1024ColonsMustBeSpacedCorrectly.DiagnosticId))
+                {
+                    continue;
+                }
+
+                SyntaxToken token = root.FindToken(diagnostic.Location.SourceSpan.Start);
+                if (token.IsMissing)
+                {
+                    continue;
+                }
+
+                context.RegisterCodeFix(CodeAction.Create(SpacingResources.SA1024CodeFix, t => GetTransformedDocumentAsync(context.Document, root, token)), diagnostic);
+            }
+        }
+
+        private static Task<Document> GetTransformedDocumentAsync(Document document, SyntaxNode root, SyntaxToken token)
+        {
+            bool requireBefore = true;
+            switch (token.Parent.Kind())
+            {
+                case SyntaxKind.LabeledStatement:
+                case SyntaxKind.CaseSwitchLabel:
+                case SyntaxKind.DefaultSwitchLabel:
+                // NameColon is not explicitly listed in the description of this warning, but the behavior is inferred
+                case SyntaxKind.NameColon:
+                    requireBefore = false;
+                    break;
+            }
+
+            // check for a following space
+            bool missingFollowingSpace = true;
+            if (token.HasTrailingTrivia)
+            {
+                if (token.TrailingTrivia.First().IsKind(SyntaxKind.WhitespaceTrivia))
+                {
+                    missingFollowingSpace = false;
+                }
+                else if (token.TrailingTrivia.First().IsKind(SyntaxKind.EndOfLineTrivia))
+                {
+                    missingFollowingSpace = false;
+                }
+            }
+
+            bool hasPrecedingSpace = token.HasLeadingTrivia;
+            if (!hasPrecedingSpace)
+            {
+                // only the first token on the line has leading trivia, and those are ignored
+                SyntaxToken precedingToken = token.GetPreviousToken();
+                SyntaxTriviaList triviaList = precedingToken.TrailingTrivia;
+                if (triviaList.Count > 0 && !triviaList.Last().IsKind(SyntaxKind.MultiLineCommentTrivia))
+                {
+                    hasPrecedingSpace = true;
+                }
+            }
```

Comment:
The code above makes a big assumption that trivia between tokens on the same line are always placed as trailing trivia of the first token. An alternative would be to merge the trailing and leading trivia of the two tokens and do analysis on that
```C#
    var combinedTrivia = precedingToken.TrailingTrivia.AddRange(token.LeadingTrivia);
```